### PR TITLE
Adding homepage header and cards seeds

### PR DESCRIPTION
**Pull request by PierreMesure**
References
===================
- Related issue: #2667

Objectives
===================
- Add dev seeds to customize the homepage

Visual Changes
===================
Before:

screenshot-2018-6-13 consul 1

After:

screenshot-2018-6-13 consul

Notes
===================
- I added images in the dev_seeds folder, not sure if that is good practice.
- Also not sure about the way I'm creating the image attachments for the widgets. It works but is it the best way?

### CODE REVIEW

**Comment by PierreMesure** [comment]
Also these comments:
- the title and the label of a card don't appear if there's no picture
- the text of the 'Save card' button changes to `<span class=&quot;translation_missing&quot; title=&quot;translation missing: en.admin.homepage.update.submit_card&quot;>Submit Card</span>`when there is an error in the card form
- in the card form, the image is displayed cropped and enlarged for no reason
- (design) the card description is underlined on hover, which makes it less readable. Wouldn't it be better to design a hover state for the whole card if it is meant to be clickable?

File: Gemfile.lock [hunk 1]
```diff
@@ -41,7 +41,7 @@ GEM
     acts_as_votable (0.11.1)
     addressable (2.5.2)
       public_suffix (>= 2.0.2, < 4.0)
-    ahoy_matey (1.6.0)
```
Comment (aitbw) [inline Gemfile.lock]: Why did you update this file? I know most of them all minor updates but this PR should be focused on **only** adding seeds to populate the homepage

File: db/dev_seeds/widgets.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  Widget::Card.create!(
+    title: 'How do participatory budgets work?',
+    description: " Participatory budgets allow citizens to propose and decide directly how to spend part of the budget, with monitoring and rigorous evaluation of proposals by the institution. Maximum effectiveness and control with satisfaction for everyone.",
+    link_text: 'More about Participatory budgets',
+    link_url: 'https://youtu.be/igQ8KGZdk9c',
+    label: 'Participatory budgets',
+    header: FALSE,
+    image_attributes: create_image_attachment('budget')
+  )
+end
```
Comment (aitbw) [inline db/dev_seeds/widgets.rb]: Could you please add a trailing whitespace at the end of this file?

File: lib/tasks/homepage.rake [hunk 3]
```diff
@@ -10,5 +10,4 @@ namespace :homepage do
       Setting['feature.homepage.widgets.feeds.processes'] = true
     end
   end
-
 end
```
Comment (aitbw) [inline lib/tasks/homepage.rake]: Could you please add a trailing whitespace at the end of this file?

**Comment by aitbw** [comment]
Hi Pierre! :wave: Thanks for collaborating once again with CONSUL!

Please, take your time to review my comments. Feel free to ask anything before updating your PR :relaxed:

File: db/dev_seeds/widgets.rb (hunk 2 above)
Comment (PierreMesure) [reply db/dev_seeds/widgets.rb]: Done

File: lib/tasks/homepage.rake (hunk 3 above)
Comment (PierreMesure) [reply lib/tasks/homepage.rake]: Done

File: db/dev_seeds/widgets.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
+section "Creating header and cards for the homepage" do
+
+  def create_image_attachment(type)
+    {
+      cached_attachment: File.new("db/dev_seeds/images/#{type}_background.jpg"),
+      title: "#{type}_background.jpg",
+      user: User.where(id: 1).first
```
[the remark this answers is not in the document]
Comment (PierreMesure) [reply db/dev_seeds/widgets.rb]: Done

File: db/dev_seeds/widgets.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
+section "Creating header and cards for the homepage" do
+
+  def create_image_attachment(type)
+    {
+      cached_attachment: File.new("db/dev_seeds/images/#{type}_background.jpg"),
```
[the remark this answers is not in the document]
Comment (PierreMesure) [reply db/dev_seeds/widgets.rb]: Done

**Comment by PierreMesure** [comment]
Thanks, @aitbw. I applied the changes you recommended.

Can you confirm that db/dev_seeds/images is a good location for images?

@decabeza, I chose images from Unsplash but they are quite random. Feel free to give feedback about that.

@voodoorai2000 and @decabeza, since you were the one to implement #2641, what do you think about the points I mentioned up there?

**Comment by aitbw** [comment]
Great @PierreMesure ! :tada:

To complete my review, could you please drop commits `9d88e5f` and `f698471` and fixup commit `9cf089a`? This is to exclude the `Gemfile.lock`-related commits from this PR and to have just 1 commit for this proposed change —this is because:

1) To keep Git history as clean as possible
2) Such a small change does not require several commits in order to complete an issue

Let me know if you need anything else :relaxed:

**Comment by voodoorai2000** [comment]
@PierreMesure this is great! Thank you so much 😌 
Sure, `db/dev_seeds/images` is good 👍 
Thanks for noting the other polishing details 😌we'll create new issues to fix them ;)

**Comment by PierreMesure** [comment]
@aitbw Thanks for your review. I reset the commits instead of squashing them because I was struggling to force push it. I agree it looks cleaner but I'm not sure it's a good practice for bigger changes where the history of a branch can be useful.

**Comment by aitbw** [comment]
Thanks @PierreMesure ! :tada: 

And yes, you are correct: for bigger PRs it is not recommended doing that but for this one, it's better to keep it as clean and concise as possible :relieved:

**Comment by PierreMesure** [comment]
I messed with the CI by deleting some commits but the last build [succeeded](https://travis-ci.com/PierreMesure/consul/builds/76404397) on my fork.

**Comment by PierreMesure** [comment]
@aitbw, @voodoorai2000 Is there anything more I need to do before it gets merged?

**Comment by decabeza** [comment]
Hi @PierreMesure! sorry for the delay on the response :pray:

Don't worry about random images, are ok! :wink:

I created a [new issue](https://github.com/consul/consul/issues/2691) with all your comments! 

Many thanks! :relieved:

**Pull request merged**